### PR TITLE
Fix NPE when secret does not exist

### PR DESCRIPTION
**Pull request by sknot-rh**
Signed-off-by: Stanislav Knot <sknot@redhat.com>

### Type of change
- Bugfix

### Description
Fixes https://github.com/strimzi/strimzi-kafka-operator/issues/5897
While working on https://github.com/strimzi/strimzi-kafka-operator/pull/5549 I forgot to add a null check.

### Checklist

- [ ] Write tests
- [ ] Make sure all tests pass
- [ ] Update documentation
- [ ] Check RBAC rights for Kubernetes / OpenShift roles
- [ ] Try your changes from Pod inside your Kubernetes and OpenShift cluster, not just locally
- [ ] Reference relevant issue(s) and close them after merging
- [ ] Update CHANGELOG.md
- [ ] Supply screenshots for visual changes, such as Grafana dashboards

### CODE REVIEW

**Review by scholzj — APPROVED** [review]
Is it worth having some UT for this?

File: operator-common/src/main/java/io/strimzi/operator/common/Util.java [hunk 1]
```diff
@@ -585,7 +585,7 @@ public static String sha1Prefix(String toBeHashed)   {
     private static Future<Integer> addSecretHash(SecretOperator secretOperations, String namespace, GenericSecretSource genericSecretSource) {
         if (genericSecretSource != null) {
             return secretOperations.getAsync(namespace, genericSecretSource.getSecretName())
-                    .compose(secret -> Future.succeededFuture(secret.getData().get(genericSecretSource.getKey()).hashCode()));
+                    .compose(secret -> secret == null ? Future.failedFuture("Secret " + genericSecretSource.getSecretName() + " not found") : Future.succeededFuture(secret.getData().get(genericSecretSource.getKey()).hashCode()));
```
Comment (tombentley) [inline operator-common/src/main/java/io/strimzi/operator/common/Util.java:588]: I think a block-style lamda with an `if` would be easier to read.

**Review by tombentley — APPROVED** [review]

**Comment by tombentley** [comment]
@sknot-rh are you going to merge this?

**Pull request merged**